File: java/src/Ice/src/main/java/com/zeroc/IceInternal/ResponseHandler.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package com.zeroc.IceInternal;

public interface ResponseHandler {
  void sendResponse(int requestId, com.zeroc.Ice.OutputStream os, byte status, boolean amd);

  void sendNoResponse();

  void dispatchException(
      int requestId, com.zeroc.Ice.LocalException ex, int requestCount, boolean amd);
}

File: csharp/src/Ice/Internal/ResponseHandler.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.Internal;

public interface ResponseHandler
{
    void sendResponse(int requestId, Ice.OutputStream os, byte status, bool amd);
    void sendNoResponse();
    void dispatchException(int requestId, Ice.LocalException ex, int requestCount, bool amd);
}
